### DISCUSSION TITLE
"Query Error: Invalid Signature" when querying a Postgres RDS Database in AWS

### DISCUSSION THREAD

**Discussion by cnrodrig** [Budibase/budibase#10100] [Help]
Hi!

I just created a connection to a Postgres DB in AWS. When i create a simple query like select count(*) from table and press run query i received the message "Query Error: Invalid Signature"

Please help!

**Comment by ConorWebb96** [comment]
Hey @cnrodrig,

There isn't a lot of information to go on here.

I suspect that your error is an issue with the authentication or authorisation of your PostgreSQL RDS instance on AWS.

Below I've provided a few steps you can use to possibly debug this further.

1. Check your database credentials: Make sure you are using the correct username and password to connect to your PostgreSQL database. You can verify this by logging in to your AWS console and checking the RDS.
2. Verify your connection string: Ensure that your connection string is correctly formatted and includes the correct host, port, database name, and credentials.
3. Check your security group settings: Ensure that your RDS instance's security group allows incoming traffic from the IP address or range of IP addresses that you are using to connect to the database. You can verify this by going to the RDS instance details and checking the inbound rules of the associated security group.
4. Verify SSL settings: If you are using SSL to connect to your PostgreSQL database, make sure that your SSL settings is configured properly.
5. Check your IAM roles and permissions: Ensure that your AWS IAM roles have the correct permissions to access your RDS.
6. Review RDS logs: Check the RDS logs for any error messages/warnings for additional information.

After making sure this works outside of Budibase then try to integrate into it.

I'm not quite sure if this error is Budibase related. Without additional information I'm afraid this is all I can provide to help you in this case.

I hope this helps.
